frontend/src/features/vans/vansSlice.ts

```diff
@@ -0,0 +1,78 @@
+import apiSlice from "../../app/apiSlice";
+import { type Coordinate } from "../location/locationSlice";
+
+/**
+ * A list of vans, as defined by the backend.
+ */
+export type Vans = Van[];
+
+/**
+ * A van, as defined by the backend.
+ */
+export interface Van {
+  id: number;
+  routeId: number;
+  wheelchair: boolean;
+  location?: Coordinate;
+}
+
+// --- API Definition ---
+
+type VanLocations = Record<string, Coordinate>;
+
+const vanLocationApiUrl =
+  "ws://" + process.env.EXPO_PUBLIC_API_DOMAIN + "/vans/location/subscribe/";
```

Comment:
GPT suggests using the URL builder syntax, might be a bit overkill though

// Create a new URL object for the WebSocket
const wsUrl = new URL("ws://" + baseUrl);

// Add the path to the URL
wsUrl.pathname = "/vans/location/subscribe/";

// If you have query parameters, you can add them like this:
// wsUrl.searchParams.append('param', 'value');

Reply:
do we need to switch between ws and wss depending on whether the site is https? don't really need security just want to make sure that react native browser doesn't get mad

Reply:
Oh yeah, we need to set up the wss thing.

Reply:
I think the URL is overkill though.